website/docs/crud.md

```diff
@@ -0,0 +1,293 @@
+---
+id: crud
+title: CRUD Handler
+sidebar_label: CRUD Handler
+---
+
+## CRUD
+`Core-Lib` provides `CRUD ( Create, Read, Update, Delete )` `DataAccess` classes as boilerplate code.
+
+They are as follows
+
+
+
+### CRUD
+This class is used to initialize `CRUD` with database objects, rule validators and database handlers. `CRUD` is an abstract base class for all types of `CRUDDataAccess` in `Core-Lib`.
+```python
+class CRUD(ABC):
+
+    def __init__(self, db_entity, db: SqlAlchemyDataHandlerRegistry, rule_validator: RuleValidator):
+        self._db_entity = db_entity
+        self._db = db
+        self._rule_validator = rule_validator
+```
+`db_entity` the database class or object to be passed for initialization.
+
+`db` instance of the `SqlAlchemyDataHandlerRegistry` used to connect to the database.
+
+`rule_validator` rules must be created with `RuleValidator` and used to validate data passed to the `create()` and `update()`.
+
+#### Functions provided by `CRUD`
+
+- `get(id: int)` an abstract method to be implemented with `DataAccess` subclasses
+
+
+- `delete(id: int)` an abstract method to be implemented with `DataAccess` subclasses.
+
+
+- `create(data: dict)` is used to add a new entry into the database, this function takes a `dict` with the data to be added to the database.
+```python
+def create(self, data: dict):
+```
+`data` is type `dict`, key-values pair where key is the column name and value is the entry to be added to the column.
+
+
+- `update(id: int, data: dict)` is used to update a column in database, This function takes a `dict` that contains the data to be updated 

```

Comment:
`is used to update a` TO `used to update data in the database`

it can update more then 1 column